youtubedl/gui.py

```diff
@@ -52,8 +54,22 @@ def enterURL(self):
         """
 
         link = self.lineEditURL.text()
-        self.ytube = YouTubeVideo(
-            link, progress_callback=self.download_progress)
+
+        try:
+            self.ytube = YouTubeVideo(
+                link, progress_callback=self.download_progress)
+        except pytubeExceptions.RegexMatchError:
+            # Catches a Regex Error from URLs with invalid format
+            self.showPopUp(f"{link} is an invalid URL")
+            self.logger.error("Caught error: RegexMatchError")
+            self.logger.error(f"Inputted link \"{link}\" is not a valid URL")
+            return
+        except pytubeExceptions.VideoUnavailable:
+            # Catches a VideoUnavailable Error if pytube cannot process the video
+            self.showPopUp(f"{link} does not exist")
+            self.logger.error(f"Caught error: VideoUnavailable")
+            self.logger.error(f"Inputted link \"{link}\" does not exist")
+            return
```

Comment:
That's cool. My only sugestion is to move this code to the __init__ method of the YouTubeVideo class. That way we isolate the call to 'pytubeExceptions', which is the purpose of that inherited class.

I think to do that you should remove the two calls to self.showPopUp from the block and return a string, or something, in each except clause indicating what type of error occured. In gui.py you test the link variable for its content and take the appropriate action (show a pop-up if link is an error).

Reply:
Good suggestion, but I'm not quite sure how returning the string with the error raised would work? You're welcome to change the code if it can still work as intended

Reply:
You might have to move showpopup to core.py too. And the logger.

Reply:
https://github.com/AlexPHorta/video-downloader/tree/errorfix-aph

working

Reply:
Read through the code, good job, guess my code will never reach master >:D